onnxruntime/core/providers/cpu/controlflow/loop.cc

```diff
@@ -422,7 +422,8 @@ Status LoopImpl::Execute(FeedsFetchesManager* ffm, const FeedsFetchesManager* ca
       if (graph_output_shape) {
         output_dims.reserve(graph_output_shape->dim_size() + 1);
 
-        auto dims = onnxruntime::utils::GetTensorShapeFromTensorShapeProto(*graph_output_shape);
+        const auto& tensor_shape = onnxruntime::utils::GetTensorShapeFromTensorShapeProto(*graph_output_shape);
+        const auto& dims = tensor_shape.GetDims();
```

Comment:
Aha, when reading this line, I am wondering what's the motivation of this change please. The function name (GetTensorShapeFromTensorShapeProto) may not be good though :)

Reply:
The function name seems to be good enough. It just converts TensorShapeProto to TensorShape. Reasonable enough.